pkg/translator/azurelogs/resourcelogs_to_logs.go

```diff
@@ -227,26 +218,42 @@ func copyPropertiesAndApplySemanticConventions(category string, properties *any,
 		return
 	}
 
-	// TODO: check if this is a valid JSON string and parse it?
```

Comment:
I have removed this TODO: if we parse as JSON we will lose some fields. I think it will also be worst for performance to unmarshal instead of just iterating over the map. But having `any` is an issue for performance we should look at.

Reply:
@cparkins I would like to come back to this as I think it is best to unmarshal instead of using `any`. But this will always lead to a breaking change: if we don't explicitly handle the field, we lose it. What do you think?